js/src/components/paid-ads/asset-group/asset-group.js

```diff
@@ -57,7 +55,10 @@ export default function AssetGroup( { campaign } ) {
 			/>
 			<AssetGroupSection />
 			<StepContentFooter>
-				{ ( isCreation || isEmptyAssetGroup ) && (
+				{ ( isCreation || adapter.isEmptyAssetEntityGroup ) && (
```

Comment:
It's a bit confuse now the difference between `AssetEntityGroup` and `AssetGroup` it's referring to the same?

Reply:
No, they refer to different data sources in the `CampaignAssetsForm` component.

- The state or prop named including `/AssetGroup/i` means that it relates to the form values which contain the structure of `AssetGroupFormValues` type.
- The state or prop named including `/AssetEntityGroup/i` means that it relates to the asset group data which refers to the `AssetEntityGroup` type and is obtained from the backend API.

For example, `assetGroupErrors` is calculated from the form values and it tells the validation results of the form.

Using `isEmptyAssetEntityGroup` as another example, it's calculated from the `assetEntityGroup` data and it tells whether the obtained `assetEntityGroup` data is empty or not. Its state won't be changed along with the form value updates.

If they were named `assetGroupErrors` and `isEmptyAssetGroup`, then it would be more confusing because it couldn't be distinguished they refer to which data source via their name. Furthermore, in the `CampaignAssetsForm` component, there is another variable named `emptyAssetGroup`, which refers to the empty asset group in the form values, and naming `isEmptyAssetGroup` would be more easily confused with it.

Therefore, for the components like `CampaignAssetsForm` having both data sources inside or on their props, the mid-name `Entity` is used to help distinguish the sources.

Reply:
Ok, understood. Thanks for the further explanation. It's of course good we have different naming between both cases. 

One issue I see on this approach is that is hard to know at glance the "Entity" is referring to data and without "Entity" is referring to the Form Values

Reply:
> One issue I see on this approach is that is hard to know at glance the "Entity" is referring to data and without "Entity" is referring to the Form Values

The form values have not only the asset group but also the campaign audience and budget, so the relevant variables which are used across several components were named `*AssetGroup` rather than.

I won't extremely chase finding a name that let most devs know what it's at a glance for those variables that originally have complex relations or contexts. Because it might be very subjective things just like I could say that with/without `data` or something else is also not a good mid-name for me to know whether it refers to the data source or the form values.

Names like `assetGroupFormValuesErrors`, `emptyAssetGroupFormValues` and `isEmptyAssetEntityGroup` might make it but I don't really want a wordy naming just for making it understandable at a glance but hurt the readability once devs understood the meanings through the JSDoc or the context.